adhocracy4/ratings/static/ratings/RatingBox.jsx

```diff
@@ -0,0 +1,83 @@
+import React, { useCallback, useMemo, useState } from 'react'
+import django from 'django'
+
+import { createOrModifyRating } from './rating_api'
+import RatingButton from './RatingButton'
+import config from '../../../static/config'
+
+const translations = {
+  likes: django.gettext('Likes'),
+  dislikes: django.gettext('Dislikes')
+}
+
+export const getRedirectUrl = (id) => config.getLoginUrl() + (id ? encodeURIComponent('?comment=' + id) : '')
+
+const RatingBox = ({
+  positiveRatings,
+  negativeRatings,
+  userHasRating,
+  userRating,
+  userRatingId,
+  isReadOnly,
+  contentType,
+  objectId,
+  authenticatedAs,
+  isComment,
+  render
+}) => {
+  const [ratings, setRatings] = useState({ negative: negativeRatings, positive: positiveRatings })
+  const [userRatingData, setUserRatingData] = useState({ userHasRating, userRating, userRatingId })
+
+  const clickHandler = useCallback(async (number) => {
+    if (!authenticatedAs) {
+      const redirectId = isComment ? objectId : null
+      window.location.href = getRedirectUrl(redirectId)
+    }
+
+    if (isReadOnly) {
+      return null
+    }
+
+    const [ratings, _userRatingData] = await createOrModifyRating(number, objectId, contentType, userRatingData.userRatingId)
```

Comment:
this might be a backend nitpick but I've always known the leading underscore to mark something private/internal which doesn't totally apply here, to me it would be easier to understand as `newUserRatingData` or something. More of a discussion, I'm also fine with keeping it.

Reply:
Totally makes sense, thanks!